recipes/opencascade/all/conanfile.py

```diff
@@ -0,0 +1,211 @@
+        if not self.options.shared:
+            self._cmake.definitions["BUILD_LIBRARY_TYPE"] = "Static"
+
+        self._cmake.definitions["INSTALL_DIR_BIN"] = "bin"
+        self._cmake.definitions["INSTALL_DIR_INCLUDE"] = "include"
+        self._cmake.definitions["INSTALL_DIR_LIB"] = "lib"
+        self._cmake.definitions["INSTALL_DIR_RESOURCE"] = "res/resource"
+        self._cmake.definitions["INSTALL_DIR_DATA"] = "res/data"
+        self._cmake.definitions["INSTALL_DIR_SAMPLES"] = "res/samples"
+        self._cmake.definitions["INSTALL_DIR_DOC"] = "res/doc"
+        self._cmake.definitions["INSTALL_DIR_LAYOUT"] = "Unix"
+
+        self._cmake.configure(source_folder=self._source_subfolder)
+        return self._cmake
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def _replace_package_folder(self, source, target):
+        new_name = ""
+        if os.path.isdir(os.path.join(self.package_folder, source)):
+            tools.rmdir(os.path.join(self.package_folder, target))
+            tools.rename(
+                os.path.join(self.package_folder, source),
+                os.path.join(self.package_folder, target))
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        self.copy(
+            "LICENSE_LGPL_21.txt",
+            src=self._source_subfolder,
+            dst="licenses")
+        self.copy(
+            "OCCT_LGPL_EXCEPTION.txt",
+            src=self._source_subfolder,
+            dst="licenses")
+        tools.rmdir(os.path.join(self.package_folder, "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib/cmake"))
+        if self.settings.build_type == "Debug":
+            self._replace_package_folder("libd", "lib")
+            self._replace_package_folder("bind", "bin")
+        elif self.settings.build_type == "RelWithDebInfo":
+            self._replace_package_folder("libi", "lib")
+            self._replace_package_folder("bini", "bin")
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "OpenCASCADE"
+        self.cpp_info.names["cmake_find_package_multi"] = "OpenCASCADE"
+        libs = set(tools.collect_libs(self))
+        modules = self._modules
+        modules_tks = self._modules_toolkits
+        for (index, module) in enumerate(modules):
+            self.cpp_info.components[module].libs = [
+                tk for tk in reversed(modules_tks[module])
+                if tk in libs]
+            if index > 0:
+                self.cpp_info.components[module].requires = list(modules[:index])
+
+        # 3rd-party requirements taken from https://dev.opencascade.org/doc/overview/html/index.html#intro_req_libs
+        self.cpp_info.components["Draw"].requires.extend(
+            ["tcl::tcl", "tk::tk"])
+        self.cpp_info.components["Visualization"].requires.extend(
+            ["freetype::freetype", "opengl::opengl"])
```

Comment:
OpenCascade component is more tricky:

- There is one "component" (in CMake sense) per OpenCascade module (FoundationClasses, ModelingData etc).
- There is one CMake imported target per library file.
- OpenCascade CMake imported targets are not namespaced.
- There is one `OpenCASCADE_<component>_LIBRARIES` variable per CMake component, listing associated libs

So basically for the first 2 points, you should do something like this:

```python
    package_info(self):
        # FoundationClasses targets
        self.cpp_info.components["occt_tkernel"].names["cmake_find_package"] = "TKernel"
        self.cpp_info.components["occt_tkernel"].names["cmake_find_package_multi"] = "TKernel"
        self.cpp_info.components["occt_tkernel"].libs = "TKernel"

        self.cpp_info.components["occt_tkmath"].names["cmake_find_package"] = "TKMath"
        self.cpp_info.components["occt_tkmath"].names["cmake_find_package_multi"] = "TKMath"
        self.cpp_info.components["occt_tkmath"].libs = "TKMath"

        # trick to emulate FoundationClasses component
        self.cpp_info.components["occt_foundationclasses"].names["cmake_find_package"] = "FoundationClasses"
        self.cpp_info.components["occt_foundationclasses"].names["cmake_find_package_multi"] = "FoundationClasses"
        self.cpp_info.components["occt_foundationclasses"].requires = ["occt_tkernel", "occt_tkmath"]
```

Reply:
I see. It's weird that their export of CMake targets doesn't reflect those internal dependencies. It means their export is broken.